src/modules/admin/components/users/UserAuditPage.tsx

```diff
@@ -1,25 +1,134 @@
+import { AdminDashboardRoutes } from '@/routes/routes';
+import { generateSafePath } from '@/utils/pathEncoding';
+
+type EntityType = 'clients' | 'enrollments';
 
-const UserAuditPage: React.FC = () => {
+const toggleItems: ToggleItem<EntityType>[] = [
+  {
+    value: 'clients',
+    label: 'Clients',
+    Icon: ClientIcon,
+  },
+  {
+    value: 'enrollments',
+    label: 'Enrollments',
+    Icon: EnrollmentIcon,
+  },
+];
+
+interface Props {
+  entityType: EntityType;
+}
+const UserAuditPage: React.FC<Props> = ({ entityType }) => {
   const { userId } = useSafeParams() as { userId: string };
   const { user, loading } = useUser(userId);
+  const [search, setSearch, debouncedSearch] = useDebouncedState<string>('');
+  const navigate = useNavigate();
+  //const defaultStartDate = useMemo<string>(() => {
+  //  const today = new Date();
+  //  const twoWeeksAgo = subWeeks(today, 2);
+  //  return formatISO(twoWeeksAgo);
+  //}, []);
 
   if (!user && loading) return <Loading />;
   if (!user) return <NotFound />;
 
-  // TODO: support toggling between Access History and Edit History
+  const handleTabChange = (value: EntityType) => {
+    switch (value) {
+      case 'clients':
+        navigate(
+          generateSafePath(AdminDashboardRoutes.USER_CLIENT_ACCESS_HISTORY, {
+            userId,
+          })
+        );
+        break;
+      case 'enrollments':
+        navigate(
+          generateSafePath(
+            AdminDashboardRoutes.USER_ENROLLMENT_ACCESS_HISTORY,
+            { userId }
+          )
+        );
+        break;
+    }
+  };
+
   return (
     <>
       <PageTitle title={`Access History for ${user.name}`} />
       <Paper>
-        <UserAccessHistoryTable userId={userId} />
+        <Box mt={2} px={2}>
+          <Typography variant='subtitle1' sx={{ mb: 2 }}>
+            {`Data on this page may be delayed by up to an hour. Recent user activity may not be immediately visible.`}
+          </Typography>
+          <Typography sx={{ fontWeight: 600, mb: 0.5 }}>
+            View user access by
+          </Typography>
+          <CommonToggle
+            sx={{ mb: 3 }}
+            value={entityType}
+            onChange={handleTabChange}
+            items={toggleItems}
+            size='small'
+            variant='gray'
+          />
+          {entityType == 'clients' && (
+            <CommonSearchInput
+              label='Search client access'
+              name='searchClients'
+              placeholder='Search by client name/ID'
+              value={search}
+              onChange={setSearch}
+              fullWidth
+              size='medium'
+              searchAdornment
+            />
+          )}
+          {entityType == 'enrollments' && (
+            <CommonSearchInput
+              label='Search enrollment access'
+              name='searchEnrollments'
+              placeholder='Search by enrollment name/ID, client name/ID'
```

Comment:
"enrollment name" is a little misleading since you can't actually search by project name. As commented on the baackend PR, I think that's fine but we should have a project filter option.

Maybe:
```suggestion
              placeholder='Search by Enrollment ID, Client ID, or Client Name'
```

Reply:
good catch, I made the language more similar to the client search